src/python/pants/backend/python/target_types_test.py

```diff
@@ -364,35 +366,109 @@ def test_parse_requirements_file() -> None:
     }
 
 
-def test_python_distribution_dependency_injection() -> None:
+def test_inject_python_distribution_dependencies() -> None:
     rule_runner = RuleRunner(
         rules=[
             inject_python_distribution_dependencies,
+            resolve_pex_entry_point,
+            resolve_python_distribution_entry_points,
+            *import_rules(),
+            *python_sources.rules(),
             QueryRule(InjectedDependencies, [InjectPythonDistributionDependencies]),
         ],
-        target_types=[PythonDistribution, PexBinary],
+        target_types=[PythonDistribution, PythonRequirementLibrary, PythonLibrary, PexBinary],
         objects={"setup_py": PythonArtifact},
     )
+    rule_runner.add_to_build_file(
+        "",
+        dedent(
+            """\
+            python_requirement_library(
+                name='ansicolors',
+                requirements=['ansicolors'],
+                module_mapping={'ansicolors': ['colors']},
+            )
+            """
+        ),
+    )
+    rule_runner.create_file("project/app.py")
     rule_runner.add_to_build_file(
         "project",
         dedent(
             """\
-            pex_binary(name="my_binary")
+            pex_binary(name="my_binary", entry_point="who_knows.module:main")
+
+            python_library(name="my_library", sources=["app.py"])
+
             python_distribution(
-                name="dist",
+                name="dist-a",
                 provides=setup_py(
-                    name='my-dist'
+                    name='my-dist-a'
                 ).with_binaries({"my_cmd": ":my_binary"})
             )
+
+            python_distribution(
+                name="dist-b",
+                provides=setup_py(
+                    name="my-dist-b"
+                ),
+                entry_points={
+                    "console_scripts":{
+                        "b_cmd": "project.app:main",
+                        "cmd_2": "//project:my_binary",
+                    }
+                },
+            )
+
```

Comment:
(In a followup) Could you please add a test for dep inference of entry points defined in the `provides=setup_py()`?

Reply:
Sure thing. But trying to remember what's changed since: https://github.com/pantsbuild/pants/pull/11872#discussion_r613511687
> [...] you can still use the old setup_py() way of doing it, only you'll miss out a) dependency inference

Hmm... looking at the `inject_python_distribution_dependencies` I don't think that we infer dependencies on entry_points from `setup_py()` at all.. so that test should not work ;)

Reply:
But then again, that was the point of #11807 which supposedly was incorporated into this PR... eeh.. a little too much water under the bridge from start to finnish, I'd say..

Reply:
So, question becomes, do we want to spend time fixing that dep inference, or simply leave be waiting for #12410

Reply:
Ah, okay. Makes sense.

Totally depends on what you have the time and interest to do: #12410 is more of a blocker b/c right now we have two Pants-provided ways of doing something, and we want to get rid of the `.with_binaries()` tech debt.

It would also be great to add dep inference for entry points defined inside a `setup_py()`, but that isn't a blocker. It's a new feature. If you have the time and interest, that'd be great - okay if not. Either way, maybe create a issue to track this so we don't forget it's a feature request?

Reply:
I'll post a PR deprecating `.with_binaries()` tomorrow.
Can also look into the dep inference on setup_py entry_points, shouldn't be too difficult, now :)

Reply:
Deprecation PR: #12413

Reply:
Dep inference for `setup_py()` entry points PR: #12414